collection-scripts/gather_profiling_kubelet

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="${BASE_COLLECTION_PATH:-must-gather}"
+
+PROFILING_KUBELET_LOG_PATH="${BASE_COLLECTION_PATH}/pprof/kubelet"
+
+
+PROFILING_KUBELET_PROXY=""
+PROFILING_KUBELET_DURATION_SEC=30
+
+_gather_profiling_kubelet_stop_proxy() {
+    kill "$PROFILING_KUBELET_PROXY" > /dev/null 2>&1
+    PROFILING_KUBELET_PROXY=""
+}
+
+# _gather_profiling_kubelet_node_exists $NODE
+# returns "true" if $NODE is found among the cluster nodes, "false" otherwise.
+_gather_profiling_kubelet_node_exists() {
+    local node="$1"
+
+    oc get node "$node" > /dev/null 2>&1
+    return "$?"
+}
+
+# _gather_profiling_kubelet_start_proxy
+# returns "true" if the kubernetes apiserver can be reached via a proxy, returns "false" otherwise.
+_gather_profiling_kubelet_start_proxy() {
+    local rcode
+
+    oc proxy > /dev/null 2>&1 & PROFILING_KUBELET_PROXY=($!)
+    # be sure the proxy is started and working
+    for i in `seq 1 10`; do
+        sleep 1
+        curl http://localhost:8001/api > /dev/null 2>&1
+        rcode="$?"
+
+        [ "$rcode" = 0 ] && return 0
+    done
+    return 1
+}
+
+_gather_profiling_kubelet_collect_data() {
+    local node="$1"
+    local output_base="${PROFILING_KUBELET_LOG_PATH}/${node}"
+    local output_heap="${output_base}_heap.out"
+    local output_prof="${output_base}_prof.out"
+    local url_base="http://localhost:8001/api/v1/nodes/${node}/proxy/debug/pprof"
+
+    curl -o "$output_prof" "${url_base}/profile?seconds=${PROFILING_KUBELET_DURATION_SEC}" > /dev/null 2>&1
```

Comment:
Should we swap this out with `oc raw` calls?

Reply:
Thanks for the hint, moved to oc get --raw (and so we don't even need the proxy)!